scarlet/component.py

```diff
@@ -350,30 +365,55 @@ class CubeComponent(Component):
         Hyper-spectral bounding box
     """
 
-    def __init__(self, frame, cube, bbox=None):
+    def __init__(self, frame, bbox, cube):
         parameters = (cube,)
-        super().__init__(frame, *parameters, bbox=bbox)
+        super().__init__(frame, bbox, *parameters)
 
     @property
     def cube(self):
-        return self._pad_cube(self._parameters[0]._data)
+        return self._parameters[0]._data
 
-    def get_model(self, *parameters):
+    def get_model(self, *parameters, frame=None):
         cube = None
         for p in parameters:
             if p._value is self._parameters[0]:
-                cube = self._pad_cube(p)
+                cube = p
 
         if cube is None:
             cube = self.cube
-
+        if frame is not None:
+            cube = self.project(cube, frame)
         return cube
 
-    def _pad_cube(self, cube):
-        if self.bbox is not None:
-            padded = np.pad(cube, self.pad_width, mode="constant", constant_values=0)
-            return padded[self.slices]
-        return cube
+
+@primitive
+def _add_models(*models, full_model, slices):
+    """Insert the models into the full model
+    """
+    for i in range(len(models)):
+        if hasattr(models[i], "_value"):
+            full_model[slices[i][0]] += models[i][slices[i][1]]._value
```

Comment:
The definition of the tuple `slices` needs to be explained in the docstrings, in particular their order.

Reply:
Or at least as an inline comment for us